doc/classes/ProjectSettings.xml

```diff
@@ -1567,9 +1567,10 @@
 		<member name="rendering/gles3/shaders/shader_compilation_mode" type="int" setter="" getter="" default="0">
 			If set to [code]Asynchronous[/code] and available on the target device, asynchronous compilation of shaders is enabled (in contrast to [code]Asynchronous[/code]).
 			That means that when a shader is first used under some new rendering situation, the game won't stall while such shader is being compiled. Instead, a fallback will be used and the real shader will be compiled in the background. Once the actual shader is compiled, it will be used the next times it's used to draw a frame.
-			Depending on the async mode configured for a given material/shader, the fallback will be an "ubershader" (the default) or just skip rendering any item it is applied to.
+			Depending on the [member SpatialMaterial.async_mode] mode configured for a given material, the fallback will be an "ubershader" (the default) or just skip rendering any item it is applied to. In custom [ShaderMaterials], the async mode is set using [code]render_mode async_visible;[/code] (default) or [code]render_mode async_hidden;[/code] at the top of the shader.
 			An ubershader is a very complex shader, slow but suited to any rendering situation, that the engine generates internally so it can be used from the beginning while the traditional conditioned, optimized version of it is being compiled.
 			To reduce loading times after the project has been launched at least once, you can use [code]Asynchronous + Cache[/code]. This also causes the ubershaders to be cached into storage so they can be ready faster next time they are used (provided the platform provides support for it).
+			[b]Note:[/b] Asynchronous compilation requires driver support for the [code]GL_ARB_get_program_binary[/code] OpenGL extension. This extension is supported by all hardware that supports OpenGL 4.1. This includes NVIDIA Fermi and later, AMD GCN 1.0 and later, Intel Haswell and later, and Apple Silicon GPUs.
```

Comment:
Just a point for discussion. GPUInfo only has 24 reports of OpenGL 3.3 capable devices not supporting the GL_ARB_get_program_binary and in all those cases it looks like out of date drivers are responsible (they are all on Windows 7, Windows XP, or using Parellels to run Windows 10 on macOS).

So we could consider saying something like "this extension should be supported by all hardware that supports OpenGL 3.3"

Reply:
> So we could consider saying something like "this extension should be supported by all hardware that supports OpenGL 3.3"

In practice, that's not true in all situations, which is why I'd prefer to mention OpenGL 4.1 as a requirement.

Reply:
Well, its actually the opposite. In practice it is true. In theory there can be hardware that supports OpenGL 3.3 but not the extension, but in practice everything that actually supports OpenGL 3.3 does support the extension. 

That being said, perhaps a better middle ground would be to soften the current language. OpenGL 4.1 support is definitely not a requirement, so we shouldn't lead users to think that. Maybe something like:
```suggestion
			[b]Note:[/b] Asynchronous compilation requires driver support for the [code]GL_ARB_get_program_binary[/code] OpenGL extension. This extension is supported by all hardware that supports OpenGL 4.1 or higher as well as most hardware that supports OpenGL 3.3 or higher.
```